File: src/models/Stock.ts
export type NewStock={
      code:string,
      date:string,
      amount:number,
      unitPrice:number
}
export type StockApi={
    change:number,
    close:number,
    log:string
    market_cap:number,
    name:string,
    sector:string,
    stock:string,
    type:string,
    volume:number

}

export class MyStock {
    public id: number;
    public code: string;
    public unitPrice: number;
    public amount: number;
    public mediumPrice: number;
    public earnings: number;
    public value: number;
    public cost: number;

    public constructor(id: number, code: string, unitPrice: number, amount: number, mediumPrice: number, earnings: number) {
        this.id = id;
        this.code = code;
        this.unitPrice = unitPrice;
        this.amount = amount;
        this.value = this.unitPrice * amount;
        this.mediumPrice = mediumPrice;
        this.cost = this.mediumPrice * amount;
        this.earnings = earnings;
    }

    public AddNew(amount: number, unitPrice: number, oldCost: number, amountOld: number) {
        const buyCost = amount * unitPrice;
        const newCost = buyCost + oldCost;
        const newAmount = amount + amountOld;
        const average = newCost / newAmount;

        this.amount = newAmount;
        this.mediumPrice = average;
        this.cost = newCost;
    }
}

File: stockz-bucketz-api/Models/Stock.cs
using System.ComponentModel.DataAnnotations;

namespace stockz_bucketz_api.Models
{
    public class Stock
    {
        public Stock(string idUser, string code, int amount, double value, double mediumPrice, double earnings, double unitPrice)
        {
            IdUser = idUser;
            Code = code;
            Amount = amount;
            Value = value;
            MediumPrice = mediumPrice;
            Earnings = earnings;
            UnitPrice = unitPrice;
            Cost = mediumPrice * amount;
        }

        [Key]
        public int Id { get; set; }
        public string IdUser { get; set; }
        public double UnitPrice { get; set; }
        public string Code { get; set; }
        public int Amount { get; set; }
        public double Value { get; set; }
        public double Cost { get; set; }
        public double MediumPrice { get; set; }
        public double Earnings { get; set; }

        public void AddNew(int amount, double unitPrice, double oldCost, int amountOld)
        {
            var buyCost = amount * unitPrice;
            var newCost = buyCost + oldCost;
            var newAmount = amount + amountOld;
            var average = newCost / newAmount;

            this.Amount = newAmount;
            this.MediumPrice = average;
            this.Cost = newCost;
        }
    }
}
